### DISCUSSION TITLE
Expose portal so the wallet connect modal can be rendered elsewhere

### DISCUSSION THREAD

**Discussion by smblee** [family/connectkit#276] [Ideas]
Currently the modal is always rendered at root (body). This is an issue if we have wallet connect within another modal (like Radix) that has close on click outside, since the wallet connect will be recognized as "outside" (due to being at root).

https://github.com/family/connectkit/blob/main/packages/connectkit/src/components/Common/Portal/index.tsx#L24

It would be nice if we can pass in a `ref` or something to optionally render the connect modal at a different root point.

**Comment by lochie** [comment]
Hey @smblee, the current expected pattern is to pull in the `open` flag from the [`useModal` Hook](https://docs.family.co/connectkit/api-reference#hooks-usemodal) as a way to force UI to stay open if necessary.

In saying that, I agree that passing a `ref` into the `<ConnectKitProvider>` would be a good pattern to include. I'll look into this and see if this can be part of the next patch.

**Reply by smblee** [reply]
Hi @lochie Was this ever implemented?

**Reply by lochie** [reply]
Not it was not, you're welcome to make a PR to enable this though.
